### ISSUE TITLE
Introduce backward compatibility with unescaped newline characters in query

### ISSUE THREAD

**Issue by BenoitRanque** [hasura/graphql-parser-hs#50]
It seems that before version 2.0.6 was released, it was possible to make requests with an invalid JSON string as the query.
Specifically, the JSON spec requires that within strings, newlines characters must be escaped, per [RFC 4627](https://www.ietf.org/rfc/rfc4627.txt)
> All Unicode characters may be placed within the quotation marks except for the characters that **must** be escaped: quotation mark, reverse solidus, **and the control characters** (U+0000 through U+001F).

But before 2.0.6, it was possible to make requests with payloads that looked like this (note the unescaped newlines)
```
{"query":"query DataTreppPropertyUSAAddressDataTest{
  trepp_property{
    usa_address__address{
      one_line_address
    }
  }
}","variables":null,"operationName":"DataTreppPropertyUSAAddressDataTest"}
```
This is no longer possible.
~~The new, spec-compliant behavior was most likely introduced in [this](https://github.com/hasura/graphql-engine/commit/52d91e3e8d03cbe08fa82230d0f99de7ba769264) commit, which has a fairly lengthy explanation on the origin of the bug and why and how it was fixed.~~ Edit: or maybe not.

Of course most graphql clients will process query strings and produce an escaped string when converting to JSON, so most users did not notice the change.

Emphasis on most. This issue is to discuss whether we should consider introducing backwards compatibility for the previous behavior.

**Comment by abooij** [comment]
`graphql-parser-hs` is not concerned with parsing JSON. `graphql-engine` uses the [`aeson`](https://hackage.haskell.org/package/aeson) library for parsing JSON.

Indeed, it seems that aeson used to erroneously accept certain control characters in JSON strings, see [this upstream PR](https://github.com/haskell/aeson/pull/769). That PR was first released in `aeson` version 1.5.0.0. According to the git history, `graphql-engine` upgraded from `aeson` 1.4.7.1 to 1.5.4.1 during the release of `graphql-engine` v2.0.0. So the above-reported behavior is probably in place since then.

In summary:
- This behavior is not due to `graphql-parser-hs`.
- This behavior has been in place since v2.0.0 due to an upgrade of an upstream library.

Finally, as is reported above, this behavior cannot be considered a bug. I think the cleanest way to support this behavior is to have a proxy server translate the almost-JSON-syntax into proper JSON. Or just fix the graphql clients.
